server/src/main/kotlin/delta/codecharacter/server/user/UserService.kt

```diff
@@ -133,6 +162,22 @@ class UserService(
         }
     }
 
+    fun verifyReCaptcha(reCaptchaResponse: String): Boolean {
+        val url =
+            "https://www.google.com/recaptcha/api/siteverify"
+        try {
+            val requestBody = ObjectMapper().writeValueAsString(mapOf("secret" to secretKey,"response" to reCaptchaResponse))
```

Comment:
```suggestion
            val recaptchaUrl =             "https://www.google.com/recaptcha/api/siteverify?secret=${secretKey}&response=${reCaptchaResponse}"
```
- The `recaptcha-url` will be accepting the `secret` and `recaptchaCode` as `params` not as `request body`. [Check Here](https://developers.google.com/recaptcha/docs/verify)